catalog/ui/search-ui/search-endpoint/src/test/java/org/codice/ddf/ui/searchui/query/controller/SearchControllerTest.java

```diff
@@ -409,91 +489,126 @@ private void assertReplies(List<Mutable> replies) {
                     .get("format"), is("STRING"));
-            assertThat((String) typeInfo.get("metadata-content-type-version")
+            assertThat(typeInfo.get("metadata-content-type-version")
                     .get("format"), is("STRING"));
-            assertThat((String) typeInfo.get("metadata-target-namespace")
+            assertThat(typeInfo.get("metadata-target-namespace")
                     .get("format"), is("STRING"));
-            assertThat((String) typeInfo.get("resource-uri")
+            assertThat(typeInfo.get("resource-uri")
                     .get("format"), is("STRING"));
-            assertThat((Boolean) typeInfo.get("resource-uri")
+            assertThat(typeInfo.get("resource-uri")
                     .get("indexed"), is(true));
             // since resource-size is not indexed, it should be filtered out
-            assertThat((Boolean) typeInfo.get("resource-size")
+            assertThat(typeInfo.get("resource-size")
                     .get("indexed"), is(false));
-            assertThat((String) typeInfo.get("metadata")
+            assertThat(typeInfo.get("metadata")
                     .get("format"), is("XML"));
-            assertThat((String) typeInfo.get("location")
+            assertThat(typeInfo.get("location")
                     .get("format"), is("GEOMETRY"));
         }
     }
 
-    private CatalogFramework createFramework() {
-        final long COUNT = 2;
+    private CatalogFramework createPagingFramework(long count, int pagesExpected) {
+        CatalogFramework framework = mock(CatalogFramework.class);
+        List<Result> results = new ArrayList<>();
+
+        for (int i = 0; i < count * pagesExpected; i++) {
+            Result result = generateResult(i, count);
+            results.add(result);
+        }
+
+        QueryResponse firstResponse = new QueryResponseImpl(mock(QueryRequest.class),
+                new ArrayList<>(),
+                count);
+        for (int i = 0; i < count; i++) {
+            firstResponse.getResults()
+                    .add(results.get(i));
+        }
+        QueryResponse secondResponse = new QueryResponseImpl(mock(QueryRequest.class),
+                new ArrayList<>(),
+                count);
+        for (int i = (int) count; i < count * pagesExpected; i++) {
+            secondResponse.getResults()
+                    .add(results.get(i));
+        }
+
+        QueryResponse emptyResponse = new QueryResponseImpl(mock(QueryRequest.class),
+                new ArrayList<>(),
+                0);
+
+        try {
+            when(framework.query(any(QueryRequest.class))).thenReturn(firstResponse,
+                    secondResponse,
+                    emptyResponse);
+        } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+            LOGGER.debug("Error querying framework", e);
+        }
+        return framework;
+    }
+
+    private CatalogFramework createFramework(long count) {
```

Comment:
✏️ Specifying this creates a framework _mock_ would help with readability, as well as making "count" more specific. As it stands, it's rather vague what "count" is the count of. It should be immediately apparent by its name that it's the "count of results" you wish to return from a call to the catalog framework.